test/model.ts

```diff
@@ -13,38 +13,108 @@
 // limitations under the License.
 
 import * as assert from 'assert';
-import {describe, it, before, beforeEach} from 'mocha';
+import {describe, it, before, beforeEach, afterEach} from 'mocha';
+import * as sinon from 'sinon';
+import * as extend from 'extend';
 import * as proxyquire from 'proxyquire';
-import * as m from '../src/model';
-import {Dataset} from '../src/dataset';
+import * as pfy from '@google-cloud/promisify';
+import {EventEmitter} from 'events';
+import {JobOptions} from '../src/job';
+import {ServiceObject, ServiceObjectConfig, util} from '@google-cloud/common';
 
-class FakeServiceObject {
-  _calledWith: Array<unknown>;
-  constructor(...args: Array<unknown>) {
-    this._calledWith = args;
+let promisified = false;
+
+const fakePfy = Object.assign({}, pfy, {
+  promisifyAll: (c: Function) => {
+    if (c.name === 'Model') {
+      promisified = true;
+    }
+    pfy.promisifyAll(c);
+  },
+});
+
+class FakeServiceObject extends ServiceObject {
+  _calledWith: IArguments;
+  constructor(config: ServiceObjectConfig) {
+    super(config);
+    // eslint-disable-next-line prefer-rest-params
+    this._calledWith = arguments;
   }
 }
 
+let isCustomTypeOverride: Function | null;
+const fakeUtil = Object.assign({}, util, {
+  isCustomType: (...args: Array<{}>) => {
+    return (isCustomTypeOverride || util.isCustomType)(...args);
+  },
+  noop: () => {},
+});
+
+const sandbox = sinon.createSandbox();
+
 describe('BigQuery/Model', () => {
   const MODEL_ID = 'my_model';
-  const DATASET = {id: 'my_dataset'} as Dataset;
 
-  let Model: typeof m.Model;
-  let model: m.Model;
+  const DATASET = {
+    id: 'dataset-id',
+    createTable: util.noop,
+    bigQuery: {
+      projectId: 'project-id',
+      job: (id: string) => {
+        return {id};
+      },
+      apiEndpoint: 'bigquery.googleapis.com',
+      request: util.noop,
+    },
+  };
 
   before(() => {
     Model = proxyquire('../src/model.js', {
       '@google-cloud/common': {
         ServiceObject: FakeServiceObject,
+        util: fakeUtil,
       },
+      '@google-cloud/promisify': fakePfy,
     }).Model;
+
+    const modelCached = extend(true, {}, Model);
+
+    // Override all util methods, allowing them to be mocked. Overrides are
```

Comment:
Not really following what's happening here super closely, but generally I rely on `sinon` for doing this kinda thing